src/ptc/start.clj

```diff
@@ -75,43 +75,49 @@
 
 (defn listen-and-consume-from-queue
   "Listen to QUEUE on CONNECTION for messages,
-  and call (TASK! message) until it is false."
-  ([task! connection queue]
-   (loop [counter 0]
-     (if-let [peeked (jms/ednify (peek-message connection queue))]
-       (do (log/infof "Peeked message %s: %s" counter peeked)
-           (if (task! peeked)
-             (let [consumed (jms/ednify (consume connection queue))]
-               (log/infof "Task complete, consumed message %s" counter)
-               (if (not (= peeked consumed))
-                 (log/warnf
-                  (str/join \space ["Messages differ:"
-                                    (pprint (data/diff peeked consumed))])))
-               (recur (inc counter)))
-             (do
-               (log/errorf
-                (str/join
-                 \space ["Task returned nil/false,"
-                         "not consuming message %s and instead exiting"])
-                counter)
-               peeked)))
-       (recur counter))))
-  ([connection queue]
-   (listen-and-consume-from-queue identity connection queue)))
+  and call (TASK! message) with PUSH-TO param
+  until it is false."
+  ([task! connection queue push-to]
+   (letfn [(task-push-to! [msg] ((partial task! push-to) msg))]
```

Comment:
Sure, this makes sense in terms of letting the `task!` deal with the parameters rather than passing the arguments all the way from `main` top down to the message loop. I initially did not like about this since this makes what env vars PTC reads from system implicit.

Reply:
I don't think it makes it any more or less implicit - you still have to read the code to know what environment variables are being read. This way, you have to pass the task and then bucket together when you could simply pass one closure.

Reply:
Yup, I think I like your suggestion which makes most other irrelevant functions know less. Applying it now.